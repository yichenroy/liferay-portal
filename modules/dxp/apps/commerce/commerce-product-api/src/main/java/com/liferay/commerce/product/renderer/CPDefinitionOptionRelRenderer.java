/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.renderer;

import aQute.bnd.annotation.ProviderType;

import com.liferay.portal.kernel.exception.PortalException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author Marco Leo
 */
@ProviderType
public interface CPDefinitionOptionRelRenderer {

	public String render(
		long cpDefinitionId, HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse) throws PortalException;

}