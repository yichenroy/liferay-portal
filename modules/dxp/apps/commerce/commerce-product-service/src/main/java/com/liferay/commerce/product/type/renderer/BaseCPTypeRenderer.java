/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.type.renderer;

import com.liferay.commerce.product.type.CPTypeRenderer;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.PortletConstants;
import com.liferay.portal.kernel.portlet.PortletBag;
import com.liferay.portal.kernel.portlet.PortletBagPool;
import com.liferay.portal.kernel.util.Portal;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;

import java.io.IOException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
public abstract class BaseCPTypeRenderer implements CPTypeRenderer {

	public void renderJSP(
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		renderJSP(
			getServletContext(httpServletRequest), httpServletRequest,
			httpServletResponse, path);
	}

	public void renderJSP(
			ServletContext servletContext,
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		RequestDispatcher requestDispatcher =
			servletContext.getRequestDispatcher(path);

		try {
			requestDispatcher.include(httpServletRequest, httpServletResponse);
		}
		catch (ServletException se) {
			_log.error("Unable to render JSP " + path, se);

			throw new IOException("Unable to render " + path, se);
		}
	}

	public void setServletContext(ServletContext servletContext) {
		_servletContext = servletContext;
	}

	protected ServletContext getServletContext(
		HttpServletRequest httpServletRequest) {

		if (_servletContext != null) {
			return _servletContext;
		}

		String portletId = portal.getPortletId(httpServletRequest);

		if (Validator.isNotNull(portletId)) {
			String rootPortletId = PortletConstants.getRootPortletId(portletId);

			PortletBag portletBag = PortletBagPool.get(rootPortletId);

			return portletBag.getServletContext();
		}

		return (ServletContext)httpServletRequest.getAttribute(WebKeys.CTX);
	}

	@Reference
	protected Portal portal;

	private static final Log _log = LogFactoryUtil.getLog(
		BaseCPTypeRenderer.class);

	private ServletContext _servletContext;

}