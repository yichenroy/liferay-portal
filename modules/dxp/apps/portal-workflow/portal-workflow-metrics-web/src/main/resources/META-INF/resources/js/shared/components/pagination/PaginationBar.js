import {AppContext} from '../../../components/AppContext';
import DisplayResult from './DisplayResult';
import PageSizeEntries from './PageSizeEntries';
import Pagination from './Pagination';
import React from 'react';

/**
 * @class
 * @memberof shared/components
 */
class PaginationBar extends React.Component {
	render() {
		const {deltas, maxPages} = this.context;
		const {
			page,
			pageCount,
			pageSize,
			pageSizes = deltas,
			totalCount
		} = this.props;

		if (totalCount <= pageSizes[0]) {
			return <div className='pagination-bar' />;
		}

		return (
			<div className='pagination-bar'>
				<PageSizeEntries
					pageSizeEntries={pageSizes}
					selectedPageSize={pageSize}
				/>

				<DisplayResult
					page={page}
					pageCount={pageCount}
					pageSize={pageSize}
					totalCount={totalCount}
				/>

				<Pagination maxPages={maxPages} totalCount={totalCount} />
			</div>
		);
	}
}

PaginationBar.contextType = AppContext;
export default PaginationBar;
