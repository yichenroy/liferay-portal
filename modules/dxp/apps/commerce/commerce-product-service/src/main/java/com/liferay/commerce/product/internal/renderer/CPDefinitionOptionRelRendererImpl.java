/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.internal.renderer;

import com.liferay.commerce.product.model.CPDefinitionOptionRel;
import com.liferay.commerce.product.model.CPDefinitionOptionValueRel;
import com.liferay.commerce.product.renderer.CPDefinitionOptionRelRenderer;
import com.liferay.commerce.product.service.CPDefinitionOptionRelService;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderer;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderingContext;
import com.liferay.dynamic.data.mapping.model.DDMForm;
import com.liferay.dynamic.data.mapping.model.DDMFormField;
import com.liferay.dynamic.data.mapping.model.DDMFormFieldOptions;
import com.liferay.dynamic.data.mapping.model.LocalizedValue;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.util.Portal;

import java.util.List;
import java.util.Locale;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
@Component(immediate = true)
public class CPDefinitionOptionRelRendererImpl
	implements CPDefinitionOptionRelRenderer {

	@Override
	public String render(
			long cpDefinitionId, HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse)
		throws PortalException {

		DDMForm ddmForm = new DDMForm();

		Locale locale = _portal.getLocale(httpServletRequest);

		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
			_cpDefinitionOptionRelService.
				getSkuContributorCPDefinitionOptionRels(cpDefinitionId);

		for (CPDefinitionOptionRel cpDefinitionOptionRel :
				cpDefinitionOptionRels) {

			List<CPDefinitionOptionValueRel> cpDefinitionOptionValueRels =
				cpDefinitionOptionRel.getCPDefinitionOptionValueRels();

			DDMFormField ddmFormField = new DDMFormField(
				String.valueOf(cpDefinitionOptionRel.
					getCPDefinitionOptionRelId()),
				cpDefinitionOptionRel.getDDMFormFieldTypeName());

			if (!cpDefinitionOptionValueRels.isEmpty()) {
				DDMFormFieldOptions ddmFormFieldOptions =
					new DDMFormFieldOptions();

				for (CPDefinitionOptionValueRel cpDefinitionOptionValueRel :
						cpDefinitionOptionValueRels) {

					ddmFormFieldOptions.addOptionLabel(
						String.valueOf(cpDefinitionOptionValueRel.
							getCPDefinitionOptionValueRelId()),
						locale, cpDefinitionOptionValueRel.getTitle(locale));
				}

				ddmFormField.setDDMFormFieldOptions(ddmFormFieldOptions);
			}

			LocalizedValue localizedValue = new LocalizedValue(locale);

			localizedValue.addString(
				locale, cpDefinitionOptionRel.getName(locale));

			ddmFormField.setLabel(localizedValue);

			ddmFormField.setRequired(true);

			ddmForm.addDDMFormField(ddmFormField);
		}

		DDMFormRenderingContext ddmFormRenderingContext =
			new DDMFormRenderingContext();

		ddmFormRenderingContext.setContainerId("options");
		ddmFormRenderingContext.setHttpServletRequest(httpServletRequest);
		ddmFormRenderingContext.setHttpServletResponse(httpServletResponse);
		ddmFormRenderingContext.setLocale(locale);

		return _ddmFormRenderer.render(ddmForm, ddmFormRenderingContext);
	}

	@Reference
	private CPDefinitionOptionRelService _cpDefinitionOptionRelService;

	@Reference
	private DDMFormRenderer _ddmFormRenderer;

	@Reference
	private Portal _portal;

}