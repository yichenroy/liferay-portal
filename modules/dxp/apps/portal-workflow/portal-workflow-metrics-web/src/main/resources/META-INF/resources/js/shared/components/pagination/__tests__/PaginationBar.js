import PaginationBar from '../PaginationBar';
import React from 'react';
import renderer from 'react-test-renderer';
import { MockRouter as Router } from '../../../../test/mock/MockRouter';

jest.mock('../../../../components/AppContext');

test('Should render component', () => {
	const component = renderer.create(
		<Router>
			<PaginationBar page={1} pageCount={5} pageSize={5} totalCount={22} />
		</Router>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Should render component with total count minor than last page', () => {
	const component = renderer.create(
		<Router>
			<PaginationBar page={1} pageCount={5} pageSize={5} totalCount={5} />
		</Router>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});