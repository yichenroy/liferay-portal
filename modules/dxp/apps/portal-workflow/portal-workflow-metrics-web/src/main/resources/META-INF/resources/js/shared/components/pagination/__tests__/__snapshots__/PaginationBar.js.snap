// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component 1`] = `
<div
  className="pagination-bar"
>
  <div
    className="dropdown pagination-items-per-page"
  >
    <a
      aria-expanded="false"
      aria-haspopup="true"
      className="dropdown-toggle"
      data-toggle="dropdown"
      href="#1"
      role="button"
    >
      5 Entries
      <svg
        className="lexicon-icon lexicon-icon-caret-double-l"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#caret-double-l"
        />
      </svg>
    </a>
    <div
      className="dropdown-menu dropdown-menu-top"
    >
      <a
        className="dropdown-item"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        5
      </a>
      <a
        className="dropdown-item"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        10
      </a>
      <a
        className="dropdown-item"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        20
      </a>
      <a
        className="dropdown-item"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        30
      </a>
      <a
        className="dropdown-item"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        50
      </a>
      <a
        className="dropdown-item"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        75
      </a>
    </div>
  </div>
  <div
    className="pagination-results"
  >
    Showing 1 to 5 of 22 entries.
  </div>
  <ul
    className="pagination pull-right"
  >
    <li
      className="page-item disabled"
    >
      <a
        className="page-link"
        href="javascript:;"
      >
        <svg
          className="lexicon-icon lexicon-icon-angle-left"
          focusable="false"
          role="presentation"
        >
          <use
            href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-left"
          />
        </svg>
        <span
          className="sr-only"
        >
          Previous
        </span>
      </a>
    </li>
    <li
      className="page-item active"
    >
      <a
        className="page-link"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/?backPath=%2F"
        onClick={[Function]}
      >
        <svg
          className="lexicon-icon lexicon-icon-angle-right"
          focusable="false"
          role="presentation"
        >
          <use
            href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-right"
          />
        </svg>
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`Should render component with total count minor than last page 1`] = `
<div
  className="pagination-bar"
/>
`;
